taxi/forms.py

```
@@ -0,0 +1,62 @@
+import re
+
+from django import forms
+from django.contrib.auth import get_user_model
+from django.contrib.auth.forms import UserCreationForm
+
+from taxi.models import Car
+
+
+User = get_user_model()
+
+
+class DriverCreationForm(UserCreationForm):
+
+    class Meta(UserCreationForm.Meta):
+        model = User
+        fields = UserCreationForm.Meta.fields + (
+            "license_number",
+            "first_name",
+            "last_name",
+        )
+
+    def clean_license_number(self):
+        license_number = self.cleaned_data["license_number"]
+        license_verification(license_number)
+        return license_number
+
+
+class CarForm(forms.ModelForm):
+    drivers = forms.ModelMultipleChoiceField(
+        queryset=get_user_model().objects.all(),
+        widget=forms.CheckboxSelectMultiple,
+        required=False,
+    )
+
+    class Meta:
+        model = Car
+        fields = "__all__"
+
+
+class DriverLicenseUpdateForm(forms.ModelForm):
+
+    class Meta:
+        model = User
+        fields = ("license_number",)
+
+    def clean_license_number(self):
+        license_number = self.cleaned_data["license_number"]
+        license_verification(license_number)
+        return license_number
+
+
+def license_verification(license_number):
+    min_license_length = 8
+
+    if len(license_number) != min_license_length:
+        raise forms.ValidationError(
+            "License number must be exactly 8 characters.")
+    elif not isinstance(license_number, str) or not re.match(
+        r"^[A-Z]{3}[0-9]{5}$", license_number
+    ):
+        raise forms.ValidationError("Please enter correct license number.")
```

Blocking issue: `license_verification` calls `len(license_number)` before checking the value is a string. If `license_number` is None or another non-string, `len()` will raise TypeError instead of a `forms.ValidationError`. This means form validation can crash rather than returning a validation error, violating the checklist requirement: "2. Validate `license_number` while creating and updating users." Fix by checking type (e.g., `isinstance(license_number, str)`) before calling `len()` or `re.match()`, then raise `forms.ValidationError` for invalid values. See the function body starting here.